File: legacy/tests.php5/exception.php
#!/usr/bin/php

<?php

include_once ("java/Java.inc");

try {
  $here=realpath(dirname($_SERVER["SCRIPT_FILENAME"]));
if(!$here) $here=getcwd();
  $e = new java("Exception");

  // trigger ID=42
  $i = $e->inner;
  $i->o = new java("java.lang.Integer", 42);

  // should return 33
  $e->inner->meth(33);

  try {
    // should throw exception "Exception$Ex"
    $e->inner->meth(42);
    return 2;
  } catch (java_exception $exception) {
    echo "An exception occured: ".java_cast($exception, "S")."\n";

    $cause = $exception->getCause();
    echo "exception ".java_cast($cause,"S")." --> " . $cause->getID() . "\n";
    return (java_values($cause->getID()) == 42) ? 0 : 3; 
  }
} catch (exception $err) {
  print "unexpected: ".java_cast($err, "S")." \n";
  return 4;
}

File: legacy/tests.php5/Exception.java
public class Exception {

    public class Ex extends java.lang.Exception {
	int id;
	public Ex(int id) {
	    this.id = id;
	}
	public int getID() {
	    return id;
	}
    }

    public class Inner {
	public Integer o = new Integer(33);
	public Integer meth(int val) throws Ex {
	    if(val==o.intValue()) throw new Ex(o.intValue());
	    return o;
	}
    };

    public Inner inner = new Inner();
}
